x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/MountSearchableSnapshotRequest.java

```diff
@@ -234,6 +234,22 @@ public String toString() {
         FULL_COPY,
         SHARED_CACHE;
 
+        public static Storage fromString(String type) {
+            if ("full_copy".equals(type)) {
```

Comment:
should we `equalsIgnoreCase` in these branches? Or is case important?

Reply:
It's not really that important, but since `equalsIgnoreCase` doesn't take a locale so we can't pass it in, I figure it's safer to keep it as-is and be strict (we can always loosen the restrictions if we decide we need to at a later time)